invenio_rdm_records/collections/service.py

```diff
@@ -0,0 +1,118 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2024 CERN.
+#
+# Invenio-RDM is free software; you can redistribute it and/or modify
+# it under the terms of the MIT License; see LICENSE file for more details.
+"""Collections service."""
+
+from invenio_communities.proxies import current_communities
+
+from .api import Collection, CollectionTree
+
+
+class CollectionItem:
+    """Collection item."""
+
+    def __init__(self, collection: Collection):
```

Comment:
Was using type hints discussed? We don't usually use them in Invenio

Reply:
I copied some code from my prototype, thanks for spotting it 😄 

Will be removed in this PR